[Bug]: Containerd does not build for x86_64 due to cgo linking not enabled

### Problem description

When trying to run
`./build-package.sh -a x86_64 containerd`
from the latest checkout (commit 9bb77a7), the build fails with the following message:
```
android/amd64 requires external (cgo) linking, but cgo is not enabled
make: *** [Makefile:253: bin/containerd-shim] Error 1
make: *** Waiting for unfinished jobs....
android/amd64 requires external (cgo) linking, but cgo is not enabled
make: *** [Makefile:257: bin/containerd-shim-runc-v1] Error 1
```


### What steps will reproduce the bug?

1. Clone termux-packages: `git clone https://github.com/termux/termux-packages`
1. Start the build container: `./scripts/run-docker.sh`
1. From within the build container, build containerd: `./build-package.sh -a x86_64 containerd`
1. See the following output:
```
Cloning into '/home/builder/.termux-build/containerd/cache/tmp-checkout'...
remote: Enumerating objects: 5541, done.
remote: Counting objects: 100% (5541/5541), done.
remote: Compressing objects: 100% (4533/4533), done.
remote: Total 5541 (delta 1067), reused 3276 (delta 616), pack-reused 0
Receiving objects: 100% (5541/5541), 8.89 MiB | 13.16 MiB/s, done.
Resolving deltas: 100% (1067/1067), done.
Note: switching to '3dce8eb055cbb6872793272b4f20ed16117344f8'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

~/.termux-build/containerd/cache/tmp-checkout ~/.termux-build/containerd/cache
~/.termux-build/containerd/cache
Applying patch: bundle.go.patch
Applying patch: defaults_unix.go.patch
Applying patch: fix-paths.patch
Applying patch: Makefile.patch
? bin/ctr
? bin/containerd
? bin/containerd-stress
? bin/containerd-shim
? bin/containerd-shim-runc-v1
android/amd64 requires external (cgo) linking, but cgo is not enabled
make: *** [Makefile:253: bin/containerd-shim] Error 1
make: *** Waiting for unfinished jobs....
android/amd64 requires external (cgo) linking, but cgo is not enabled
make: *** [Makefile:257: bin/containerd-shim-runc-v1] Error 1
```

### What is the expected behavior?

The expected behavior is a successful compilation:
```
termux - building containerd for arch x86_64...
Building dependency libseccomp if necessary...
libseccomp@2.5.5-3 built - skipping (rm /data/data/.built-packages/libseccomp to force rebuild)
Building dependency runc if necessary...
runc@1.1.12 built - skipping (rm /data/data/.built-packages/runc to force rebuild)
Applying patch: bundle.go.patch
Applying patch: defaults_unix.go.patch
Applying patch: fix-paths.patch
Applying patch: Makefile.patch
? bin/ctr
? bin/containerd
? bin/containerd-stress
? bin/containerd-shim
? bin/containerd-shim-runc-v1
? bin/containerd-shim-runc-v2
? binaries
+ man/ctr.8
CC=gcc CFLAGS= LDFLAGS= GOOS= GOARCH= go run -mod=readonly -tags "no_btrfs" cmd/gen-manpages/main.go ctr.8 man
go: downloading github.com/urfave/cli v1.22.1
```
...
```
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/ctr'
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/containerd'
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/containerd-shim'
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/containerd-shim-runc-v2'
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/containerd-stress'
termux-elf-cleaner: Replacing unsupported DF_1_* flags 134217729 with 1 in './bin/containerd-shim-runc-v1'
INFO: READELF=llvm-readelf ... /home/builder/.termux-build/_cache/android-r26b-api-24-v4/bin/llvm-readelf
INFO: Generating symbols regex to /tmp/tmp.83roes7GT6
INFO: Done ... 0s
INFO: Total symbols 2238
INFO: Identifying files with nproc=8
INFO: Done ... 0s
INFO: Found 6 / 11 files
INFO: Running symbol checks on 6 files with nproc=8
INFO: Done ... 3s
termux - build of 'containerd' done
```

### System information

```shell
termux-info is not relevant since this is a build system issue.
```

--- a/root-packages/containerd/build.sh
+++ b/root-packages/containerd/build.sh
@@ -7,7 +7,7 @@ TERMUX_PKG_MAINTAINER="@termux"
 # $ sudo docker run -it ubuntu bash
 # docker: Error response from daemon: failed to create task for container: failed to start shim: start failed: io.containerd.runc.v2: create new shim socket: listen unix /data/data/com.termux/files/usr/var/run/containerd/s/3f71828f1d6c1ead43fded842abc9c3cf5857c74c3e0704cd83ab177e17cfe6c: bind: invalid argument: exit status 1: unknown.
 TERMUX_PKG_VERSION=1.6.21
-TERMUX_PKG_REVISION=2
+TERMUX_PKG_REVISION=3
 TERMUX_PKG_SRCURL=git+https://github.com/containerd/containerd
 TERMUX_PKG_AUTO_UPDATE=false
 TERMUX_PKG_DEPENDS="runc"
@@ -28,7 +28,7 @@ termux_step_make() {
 
 	# issue the build command
 	export BUILDTAGS=no_btrfs
-	make -j ${TERMUX_PKG_MAKE_PROCESSES}
+	SHIM_CGO_ENABLED=1 make -j ${TERMUX_PKG_MAKE_PROCESSES}
 	(unset GOOS GOARCH CGO_LDFLAGS CC CXX CFLAGS CXXFLAGS LDFLAGS
 	make -j ${TERMUX_PKG_MAKE_PROCESSES} man)
 

